publisher/internal/analytics/ethereum/types.go

```diff
@@ -1,11 +1,27 @@
 package ethereum
 
 import (
+	"encoding/json"
 	"fmt"
+	"log"
+	"reflect"
+	"strings"
+	"time"
 
+	"github.com/SyntropyNet/swapscope/publisher/pkg/analytics"
 	"github.com/SyntropyNet/swapscope/publisher/pkg/repository"
+	"github.com/SyntropyNet/swapscope/publisher/pkg/types"
 )
 
+type Operation interface {
+	// Common methods shared by Addition and Removal
+	ExtractFromEventLogs(EventLog, *Analytics) (Operation, error)
+	PrintPretty() string
+	CanPublish() bool
+	PublishToNATS(time.Time, analytics.Sender) error
+	SaveToDB(time.Time, *Analytics) error
```

Comment:
As mentioned earlier, `SaveToDB` most likely should not require a parameter to a database, however, if it is indeed required, it should definitely not depend on `Analytics`, but on an interface that implements certain methods needed by Operation in order to save some data to the database.

In that case, database parameter should be the first one to indicate the importance of it.